leetcode/339_nested_list_weight_sum.py

```diff
@@ -0,0 +1,66 @@
+from typing import List
+from collections import deque
+from __future__ import annotations
+
+# Dummy implementation to avoid errors in local IDE
+class NestedInteger:
+    def isInteger(self) -> bool:
+        """Return True if this NestedInteger holds a single integer, otherwise False."""
+        pass
+
+    def getInteger(self) -> int:
+        """Return the single integer that this NestedInteger holds, or None if it holds a nested list."""
+        pass
+
+    def getList(self) -> List[NestedInteger]:
+        """Return the nested list that this NestedInteger holds, or None if it holds a single integer."""
+        pass
+
+class Solution:
+    def depthSum(self, nestedList: List[NestedInteger]) -> int:
```

Comment:
This approach is very good. Now let's also implement it using a recursive DFS approach for practice.

Here's a few hints to get started.

1. You'll need a helper function to compute this recursively. 
2. Identify the recursive relationship. Think of the binary tree problems we did way back in DSA Week 5. What is the base case/recurrence relationship for the `depthSum(nested_list)` in relationship to the depth sum of it's children. Just describe it in words first then formulate that into a recursive helper function.